src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriterSettings.cs

```diff
@@ -802,7 +586,7 @@ private XmlWriter AddConformanceWrapper(XmlWriter baseWriter)
                 }
             }
 
-            if (this.IsQuerySpecific && (baseWriterSettings == null || !baseWriterSettings.IsQuerySpecific))
+            if (IsQuerySpecific && baseWriterSettings is not { IsQuerySpecific: true })
```

Comment:
honestly I'm not sure this is equivalent or not to the previous version (especially about null state) - I'd prefer this stayed as it was as it feels more intuitive